runtime/lua/vim/_pos.lua

```diff
@@ -0,0 +1,64 @@
+---@class vim.pos.Pos
+---@field row integer 0-based
+---@field col integer 0-based
+---@field offset? integer Only for compatibility
```

Comment:
One of the pain points with positions I noticed is to work when they are (deliberately) outside of available text (like with 'virtualedit', which might be common with blockwise selections). I.e. past the end of line.

Not sure if this fits `Pos` abstraction or methods it might be input for, though.